TCMSApp/src/client/app/runs/runs.service.js

```diff
@@ -8,8 +8,16 @@
     RunsApiService.$inject = ['$resource', 'apiUrl'];
 
     function RunsApiService($resource, apiUrl) {
+        var currentRun;
 
         return {
+            currentRun: function(_currentRun) {
+                if (!!_currentRun) {
```

Comment:
why don't you have `setCurrentRun` metehod? And call it from here, not from controller?